This corrects for implementation differences between linux lseek() and windows,
to allow large indices to work correctly.

Integrated into minimap2 in commit 5b55a607c05bdbb1ced42cc36dc95c93590c6aa1 / PR #201

diff --git a/index.c b/index.c
index 201870a..2cb13ed 100644
--- a/index.c
+++ b/index.c
@@ -504,14 +504,19 @@ mm_idx_t *mm_idx_load(FILE *fp)
 int64_t mm_idx_is_idx(const char *fn)
 {
 	int fd, is_idx = 0;
-	off_t ret, off_end;
+	int64_t ret, off_end;
 	char magic[4];
 
 	if (strcmp(fn, "-") == 0) return 0; // read from pipe; not an index
 	fd = open(fn, O_RDONLY);
 	if (fd < 0) return -1; // error
+#ifdef WIN32
+	    if ((off_end = _lseeki64(fd, 0, SEEK_END)) >= 4) {
+	    _lseeki64(fd, 0, SEEK_SET);
+#else
 	if ((off_end = lseek(fd, 0, SEEK_END)) >= 4) {
 		lseek(fd, 0, SEEK_SET);
+#endif
 		ret = read(fd, magic, 4);
 		if (ret == 4 && strncmp(magic, MM_IDX_MAGIC, 4) == 0)
 			is_idx = 1;
